tests/unit/modules/network/ovs/test_openvswitch_port.py

```diff
@@ -261,3 +261,28 @@ def test_openvswitch_port_present_runs_set_mode(self):
             commands=commands,
             test_name="test_openvswitch_port_present_runs_set_mode",
         )
+
+    def test_openvswitch_port_present_runs_set_mode_multiple(self):
+        set_module_args(
+            dict(
+                state="present",
+                bridge="test-br",
+                port="eth2",
+                tag=10,
+                external_ids={"foo": "bar"},
+                set=[
+                    "port eth2 other_config:stp-path-cost=10",
```

Comment:
If this is how this works, that's fine, but this seems oddly redundant. Is there anything stopping someone from setting properties of unrelated ports?

I would have expected something closer to

```
state="present",
bridge="test-br",
port="eth2",
tag=10,
external_ids={"foo": "bar"},
set_properties=[
    "other_config:stp-path-cost=10",
    "type=patch",
]
```

or even something like this, though even here the properties are a little hard to name

```

set_properties={
    "other_config:stp-path-cost": 10,
    "type": "patch",
}
```

I'm not expecting a change to this PR, and maybe there's some reason you want to set arbitrary properties, just want to see if this makes sense to deprecate `set` for in the future.

Reply:
@Qalthos I totally agree with you here. The way the `set` key has been designed from the beginning isn't ideal. We should deprecate and replace it with a key that only accepts structured values.

Reply:
:+1: